src/compiler/checker.ts

```diff
@@ -8321,8 +8370,8 @@ namespace ts {
             function checkForDisallowedESSymbolOperand(operator: SyntaxKind): boolean {
                 let offendingSymbolOperand =
                     someConstituentTypeHasKind(leftType, TypeFlags.ESSymbol) ? node.left :
-                    someConstituentTypeHasKind(rightType, TypeFlags.ESSymbol) ? node.right :
-                    undefined;
+                        someConstituentTypeHasKind(rightType, TypeFlags.ESSymbol) ? node.right :

```

Comment:
Changed.

I think it's worth expressing that a linting tool *should* be able to perform these sorts of checks so we can focus on the more substantial issues. Not a dig against you, just a general frustration :/